x-pack/plugins/security/server/lib/authorization/disable_ui_capabilities.ts

```diff
@@ -37,22 +37,45 @@ export function disableUICapabilitesFactory(
 
   const disableAll = (uiCapabilities: UICapabilities) => {
     return mapValues(uiCapabilities, (featureUICapabilities, featureId) =>
-      mapValues(featureUICapabilities, (enabled, uiCapability) => {
-        if (shouldDisableFeatureUICapability(featureId!, uiCapability!)) {
-          return false;
+      mapValues(featureUICapabilities, (value, uiCapability) => {
+        if (typeof value === 'boolean') {
+          if (shouldDisableFeatureUICapability(featureId!, uiCapability!)) {
+            return false;
+          }
+          return value;
         }
 
-        return enabled;
+        if (isObject(value)) {
+          return mapValues(value, () => false);
+        }
+
+        throw new Error(`Expected value type of boolean or object, but found ${value}`);
       })
     );
   };
 
   const usingPrivileges = async (uiCapabilities: UICapabilities) => {
+    function getActionsForFeatureCapability(
+      featureId: string,
+      uiCapability: string,
+      value: boolean | Record<string, boolean>
+    ): string[] {
+      if (typeof value === 'boolean') {
```

Comment:
note: this type-checking logic is repeated within this file. I made several attempts at consolidating it, but the different needs of the checks make that difficult. I'm leaving it alone for now with the hope that I'll think of something better later on